bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannAuth.java

```diff
@@ -134,49 +136,53 @@ public void setAccessToken(String newAccessToken) {
      */
     private void authorize() throws ViessmannAuthException {
         logger.debug("ViessmannAuth: State is {}: Executing step: 'authorize'", state);
-        StringBuilder url = new StringBuilder(VIESSMANN_AUTHORIZE_URL);
-        url.append("?response_type=code");
-        url.append("&client_id=").append(apiKey);
-        url.append("&code_challenge=2e21faa1-db2c-4d0b-a10f-575fd372bc8c-575fd372bc8c");
-        url.append("&redirect_uri=http://localhost:8080/viessmann/authcode/");
-        url.append("&scope=").append(VIESSMANN_SCOPE);
-        logger.trace("ViessmannAuth: Getting authorize URL={}", url);
-        String response = executeUrlAuthorize("GET", url.toString());
-        logger.trace("ViessmannAuth: Auth response: {}", response);
-        if (response != null) {
-            if (response.indexOf("<!DOCTYPE html>") >= 0) {
-                logger.warn("ViessmannAuth: Login failed. Please check user and passowrd.");
-                updateBridgeStatusLogin();
-                return;
-            }
-            if (response.indexOf("error") >= 0) {
-                logger.warn("ViessmannAuth: Login failed. Wrong code response.");
-                return;
-            }
-        }
-        try {
-            authResponse = api.getGson().fromJson(response, AuthorizeResponseDTO.class);
-            if (authResponse == null) {
-                logger.debug("ViessmannAuth: Got null authorize response from Viessmann API");
-                setState(ViessmannAuthState.NEED_AUTH);
-                return;
-            } else {
-                AuthorizeResponseDTO resp = this.authResponse;
-                if (resp == null) {
-                    logger.warn("AuthorizeResponseDTO is null. This should not happen.");
+        if (callbackUrl != null) {
+            StringBuilder url = new StringBuilder(VIESSMANN_AUTHORIZE_URL);
+            url.append("?response_type=code");
+            url.append("&client_id=").append(apiKey);
+            url.append("&code_challenge=2e21faa1-db2c-4d0b-a10f-575fd372bc8c-575fd372bc8c");
+            url.append("&redirect_uri=" + callbackUrl + "/viessmann/authcode/");
+            url.append("&scope=").append(VIESSMANN_SCOPE);
+            logger.trace("ViessmannAuth: Getting authorize URL={}", url);
+            String response = executeUrlAuthorize("GET", url.toString());
+            logger.trace("ViessmannAuth: Auth response: {}", response);
+            if (response != null) {
+                if (response.indexOf("<!DOCTYPE html>") >= 0) {
```

Comment:
```suggestion
                if (response.contains("<!DOCTYPE html>")) {
```

also below. While browsing the existing code I've seen some 
`var.indexOf(...) == -1`. `!var.contains(...)` is better style.